src/Sylius/Bundle/ApiBundle/StateProcessor/Admin/Promotion/RemoveProcessor.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\StateProcessor\Admin\Promotion;
+
+use ApiPlatform\Metadata\DeleteOperationInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException;
+use Sylius\Bundle\ApiBundle\Exception\PromotionCannotBeRemoved;
+use Sylius\Component\Core\Model\PromotionInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @template T1 of PromotionInterface
+ * @template T2
+ *
+ * @implements ProcessorInterface<T1, T2>
+ */
+final readonly class RemoveProcessor implements ProcessorInterface
```

Comment:
```suggestion
/**
 * @implements ProcessorInterface<PromotionInterface>
 */
final readonly class RemoveProcessor implements ProcessorInterface
```
This is the final class.